packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -7,11 +7,7 @@ import {parse, normalizeUrl} from '@docusaurus/utils';
 import {LoadContext} from '@docusaurus/types';
 
 export function truncate(fileString: string, truncateMarker: RegExp | string) {
-  const truncated =
-    typeof truncateMarker === 'string'
-      ? fileString.includes(truncateMarker)
-      : truncateMarker.test(fileString);
-  return truncated ? fileString.split(truncateMarker)[0] : fileString;
+  return fileString.split(truncateMarker, 1).shift()!;
```

Comment:
if you dont mind me ask - what's the ! doing?

Reply:
its a ! safe navigator.

`shift()` can return string | undefined. But it will only return undefined if array is empty. It is guaranteed that fileString.split(truncateMarker, 1) will always return a string[] with at least a length. Typescript can only infer it will return string[]

So ! is to let compiler know that we're more clever

Reply:
til